algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java

```diff
@@ -100,43 +99,52 @@ public Set<ActionableEntry> create(@NotNull CkbEntry entry, @NotNull String sour
                     sourceUrls.add("https://ckbhome.jax.org/profileResponse/advancedEvidenceFind?molecularProfileId=" + entry.profileId());
 
                     Set<String> sourceRelevantTreatmentApproaches = Sets.newHashSet();
+                    Set<String> drugClass = Sets.newHashSet();
+                    Set<String> therapy = Sets.newHashSet();
+
                     for (RelevantTreatmentApproaches relevantTreatmentApproaches : evidence.relevantTreatmentApproaches()) {
-                        DrugClass relevantTreatmentApproachesInfo = relevantTreatmentApproaches.drugClass();
+                        DrugClassInfo drugClassInfo = relevantTreatmentApproaches.drugClass();
+                        TherapyInfo therapyInfo = relevantTreatmentApproaches.therapy();
 
-                        if (relevantTreatmentApproachesInfo != null) {
-                            sourceRelevantTreatmentApproaches.add(relevantTreatmentApproachesInfo.drugClass());
+                        // If drugClassInfo contains data then therapyInfo is null. When therapyInfo contains data then drugClassInfo is null
+                        if (drugClassInfo != null) {
+                            drugClass.add(drugClassInfo.drugClass());
                         }
-                    }
 
-                    String treatmentApproachString = String.join(",", sourceRelevantTreatmentApproaches);
-                    String treatmentApproachInterpret;
-                    if (sourceRelevantTreatmentApproaches.isEmpty()) {
-                        treatmentApproachInterpret = null;
-                    } else if (treatmentApproachString.endsWith(",")) {
-                        treatmentApproachInterpret = treatmentApproachString.substring(0, treatmentApproachString.length() - 1);
-                    } else {
-                        treatmentApproachInterpret = treatmentApproachString;
+                        if (therapyInfo != null) {
```

Comment:
Maybe `else if` would be more appropriate here if these are assumed to be mutually exclusive.

Reply:
I have leave as is because this code is annotated as TODO. This should be correctly implemented in another round